src/App.tsx

```diff
@@ -17,18 +57,36 @@ export const App: React.FC = () => {
             <h1 className="title">Todos:</h1>
 
             <div className="block">
-              <TodoFilter />
+
+              <TodoFilter
+                setFilter={setCurrentFilter}
+                setSearch={setCurrentSearch}
+              />
             </div>
 
             <div className="block">
-              <Loader />
-              <TodoList />
+              {loading && <Loader />}
+
+              <TodoList
+                todos={searchedTodos}
+                setShowModal={setShowModal}
+                setOpenedTodo={setOpenedTodo}
+                openedTodo={openedTodo}
+              />
             </div>
           </div>
         </div>
       </div>
 
-      <TodoModal />
+      {loading && <Loader />}
```

Comment:
The `Loader` component is conditionally rendered twice: once inside the block at line 68 and again at line 81. This redundancy might lead to unexpected UI behavior, such as displaying two loaders simultaneously. Consider removing one of these conditional renderings to ensure the loader is displayed only once when `loading` is true.